pint/registry.py

```diff
@@ -887,6 +883,25 @@ def parse_unit_name(self, unit_name, case_sensitive=True):
                             self._suffixes[suffix],
                         )
 
+    @staticmethod
+    def _dedup_candidates(candidates):
+        """Helper of parse_unit_name.
+
+        Given an iterable of unit triplets (prefix, name, suffix), remove those with
+        different names but equal value, preferring those with a prefix.
+
+        e.g. ('kilo', 'gram', '') and ('', 'kilogram', '')
+        """
+        candidates = dict.fromkeys(candidates)  # ordered set
+        for cp, cu, cs in list(candidates):
+            assert isinstance(cp, str)
+            assert isinstance(cu, str)
+            if cs != "":
+                raise NotImplementedError("non-empty suffix")
+            if cp:
+                candidates.pop(("", cp + cu, ""), None)
+        return tuple(candidates)
+
```

Comment:
Worth noting that this function was not doing what its docstring said before; it was deduplicating only identical tuples.